lld/ELF/Writer.cpp

```diff
@@ -553,6 +553,22 @@ template <class ELFT> void Writer<ELFT>::addSectionSymbols() {
   }
 }
 
+// Returns true if the section is a data section that's read only and
+// relocatable per its section name.
+static bool isRelRoDataSection(Ctx &ctx, StringRef SectionName) {
```

Comment:
`secName`

`if (!secName.consume_front(".data.rel.ro")) return false;` then check .hot or .unlikely

This is more efficient.